the-collab-lab/tcl-24-smart-shopping-list
# 8 kr cv checked date

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



The component List was changed, an input type checkbox was added to each item. It has attributes checked and onChange. For the checked attribute, a function called isPassed24Hours is passed, this function is in charge of checking if it has been 24 hours since the user checked the box of an item, after the 24 hours pass, the checkbox will automatically uncheck. For the onChange attribute, a function called handleCheck is passed, this function is in charge of updating the lastDate attribute on firebase, only when checked is true.

## Related Issue



closes #9 

## Acceptance Criteria



[ x ] The user can click on a checkbox or a similar element from the list to mark them as purchased.
[ x ] The displayed item should be checked for 24 hours after the purchase is made (to assume the users are not gonna need the item for at least one day).

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After


 
![Captura de pantalla de 2021-04-28 16-07-11](https://user-images.githubusercontent.com/66270746/116466098-b47fb980-a833-11eb-9840-f59263c57b7a.png)


## Testing Steps / QA Criteria



From your terminal, pull down the branch
Run npm install to install the dependencies
Run npm start to view the features in the browser
Input a shared code, when you are redirected to the list view, try and check one of the items.
